### DISCUSSION TITLE
Display a PDF file in a website while preventing its download

### DISCUSSION THREAD

**Discussion by HSessien** [community/community#34134] [Other Feature Feedback, Questions, & Ideas]
I'm new to programming so my question may be silly.

I'm designing a web platform to manage PDF files. I'd like a PDF file (whose link would be put in argument) to be loaded and displayed to the user when the web page is displayed, but he wouldn't be able to download it.

I went through several options:

- embed and iframe which allow me to display it, but I can't manage to disable the download function

- dompdf with which I can't load pre-designed PDF files (my PDF files are scanned documents). I'm more interested in this method because I'd like to be able to add watermarks but I have the impression that its specialty is pdf design and not display

- With this JavaScript resource (https://github.com/Index-Zero-0/Beginners-javascript-projects/tree/main/PDF%20Viewer), I manage to display the PDFs without the download option, but I have to click on "open PDF" to choose the file to be displayed. I can't automatically load a file from my local server.

Thanks for reading this far and thanks for your help.

**Comment by jricciardi** [comment]
Hi @HSessien,

Thanks for being a part of the GitHub Community! We're glad you're here.

I wanted to jump in and mention that if you're looking for help for this specific topic, you might want to try asking for help somewhere that focuses on that topic. It's definitely possible that another GitHub user might have run into this same issue and can help, but the GitHub Community on Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck finding a solution!

**Comment by MasterInQuestion** [comment]
Impossible.
    Relevant data would be regardless sent to client.
